app/code/Magento/CatalogGraphQl/Model/Resolver/Products/DataProvider/ExtractDataFromCategoryTree.php

```diff
@@ -42,14 +49,61 @@ public function execute(\Iterator $iterator): array
             /** @var CategoryInterface $category */
             $category = $iterator->current();
             $iterator->next();
-            $nextCategory = $iterator->current();
-            $tree[$category->getId()] = $this->categoryHydrator->hydrateCategory($category);
-            $tree[$category->getId()]['model'] = $category;
-            if ($nextCategory && (int) $nextCategory->getLevel() !== (int) $category->getLevel()) {
-                $tree[$category->getId()]['children'] = $this->execute($iterator);
+
+            $pathElements = explode("/", $category->getPath());
+            $this->iteratingCategory = $category;
+
+            $currentLevelTree = $this->explodePathToArray($pathElements, self::START_CATEGORY_FETCH_LEVEL);
```

Comment:
@paliarush @nuzil 
This changes will be broke current logic

Query:
```
{
  category(id: 334) {
   id
    name
  }
}
```

The result before changes:
```
{
  "data": {
    "category": {
      "id": 334,
      "name": "Cat-1"
    }
  }
}
```

The result after changes:
```
{
  "data": {
    "category": {
      "id": 2,
      "name": null
    }
  }
}
```
The root category has always been loaded first.

Looks like we need to change 
```
const START_CATEGORY_FETCH_LEVEL = 1;
```
on
```
const START_CATEGORY_FETCH_LEVEL = 2;
```